src/pudl/transform/eia.py

```diff
@@ -1100,42 +1100,51 @@ def fix_balancing_authority_codes_with_state(
     ba_name_to_code_map.reset_index(inplace=True)
 
-    plants = plants.merge(
-        plants_entity[["plant_id_eia", "state"]],  # only merge in state, drop later
-        on=["plant_id_eia"],
-        how="left",
-        validate="m:1",
-    )
-    BACodeFix = namedtuple(
-        "BACodeFix", ["ba_code_found", "ba_code_fix", "ba_name_fix", "states"]
-    )
-    fixes = [
-        BACodeFix(
-            "PACE",
-            "PACW",
-            ba_name_to_code_map.loc[
-                ba_name_to_code_map.balancing_authority_code_eia == "PACW",
-                "balancing_authority_name_eia",
-            ].tolist()[0],
-            ["OR", "CA"],
-        ),
-        BACodeFix(
-            "PACW",
-            "PACE",
-            ba_name_to_code_map.loc[
-                ba_name_to_code_map.balancing_authority_code_eia == "PACE",
-                "balancing_authority_name_eia",
-            ].tolist()[0],
-            ["UT"],
-        ),
-    ]
-    for fix in fixes:
-        plants.loc[
-            (plants.balancing_authority_code_eia == fix.ba_code_found)
-            & (plants.state.isin(fix.states)),
-            ["balancing_authority_code_eia", "balancing_authority_name_eia"],
-        ] = [fix.ba_code_fix, fix.ba_name_fix]
-    return plants.drop(columns=["state"])
+    logger.info("Spot fixing incorrect PACW/PACE BA codes and names.")
+
+    if (
+        not ba_name_to_code_map.empty
+        and plants.balancing_authority_code_eia.isin(["PACW", "PACE"]).any()
+    ):
+        plants = plants.merge(
+            plants_entity[["plant_id_eia", "state"]],  # only merge in state, drop later
+            on=["plant_id_eia"],
+            how="left",
+            validate="m:1",
+        )
+        BACodeFix = namedtuple(
+            "BACodeFix", ["ba_code_found", "ba_code_fix", "ba_name_fix", "states"]
+        )
+        fixes = [
+            BACodeFix(
+                "PACE",
+                "PACW",
+                ba_name_to_code_map.loc[
+                    ba_name_to_code_map.balancing_authority_code_eia == "PACW",
+                    "balancing_authority_name_eia",
+                ].tolist()[0],
+                ["OR", "CA"],
+            ),
+            BACodeFix(
+                "PACW",
+                "PACE",
+                ba_name_to_code_map.loc[
+                    ba_name_to_code_map.balancing_authority_code_eia == "PACE",
+                    "balancing_authority_name_eia",
+                ].tolist()[0],
+                ["UT"],
+            ),
+        ]
+        for fix in fixes:
+            plants.loc[
+                (plants.balancing_authority_code_eia == fix.ba_code_found)
+                & (plants.state.isin(fix.states)),
+                ["balancing_authority_code_eia", "balancing_authority_name_eia"],
+            ] = [fix.ba_code_fix, fix.ba_name_fix]
+        return plants.drop(columns=["state"])
+
+    else:  # If no fixes to be made
+        return plants
```

Comment:
Rather than having a double return statement here, you can get rid of the `else:` entirely, and just make sure that no matter what, `plants` is the thing you want to return. so...

```py
... 
      for fix in fixes:
            plants.loc[
                (plants.balancing_authority_code_eia == fix.ba_code_found)
                & (plants.state.isin(fix.states)),
                ["balancing_authority_code_eia", "balancing_authority_name_eia"],
            ] = [fix.ba_code_fix, fix.ba_name_fix]
        plants = plants.drop(columns=["state"])

    return plants
```

Reply:
(I think this will quiet CodeCov down)